server/app/services/program/ProgramService.java

```diff
@@ -264,13 +264,13 @@ private CompletionStage<ProgramDefinition> syncProgramAssociations(ProgramModel
       // This method makes multiple calls to get questions for the active and
       // draft versions, so we should only call it if we're syncing program
       // associations for a draft program (which means we're in the admin flow).
-      return syncProgramDefinitionQuestions(programRepository.getProgramDefinition(program))
+      return syncProgramDefinitionQuestions(programRepository.getShallowProgramDefinition(program))
```

Comment:
The 2 callers of this function do that (getFullProgramDefinition)[https://github.com/civiform/civiform/blob/1996073c5a492934ddbb5ae22be8666d1deea625/server/app/services/program/ProgramService.java#L207] and (getFullProgramDefinitionAsync)[https://github.com/civiform/civiform/blob/1996073c5a492934ddbb5ae22be8666d1deea625/server/app/services/program/ProgramService.java#L178], so that way the syncProgramAssociations function won't even be called if the program is in the cache. Let me know if you think its worthwhile doing it here too though in case a future caller doesn't realize.

Reply:
Sounds good, thank you! In the JavaDoc it mentions the other function should be used as well.